src/lib/swapper/swappers/CowSwapper/cowBuildTrade/cowBuildTrade.ts

```diff
@@ -60,34 +62,34 @@ export async function cowBuildTrade(
     )
   }
 
-  if (buyAssetChainId !== KnownChainIds.EthereumMainnet) {
+  if (!isCowswapSupportedChainId(buyAssetChainId)) {
     return Err(
       makeSwapErrorRight({
-        message: '[cowBuildTrade] - Buy asset needs to be on ETH mainnet to use CowSwap',
+        message: '[cowBuildTrade] - Buy asset network not supported by CowSwap',
         code: SwapErrorType.UNSUPPORTED_PAIR,
         details: { buyAssetChainId },
       }),
     )
   }
+ 
+  const { assetReference: sellAssetAddress } = fromAssetId(sellAsset.assetId)
+
+  const { assetReference: buyAssetAddress } = fromAssetId(buyAsset.assetId)
 
-  const buyToken =
```

Comment:
This will break ETH buys, `buyAsset.assetId` would be `ethAssetId` for ETH, which is invalid as CoW input - we need the marker address

Reply:
Also we may want to rename the const to `COW_SWAP_NATIVE_ASSET_MARKER_ADDRESS`, looks like this is the marker address over all networks
https://github.com/gnosis/cowswap/blob/57c10c029da5c511585e729dac513451343c8557/src/custom/constants/index.ts#L73